contracts/interchain-token-service/src/interface.rs

```diff
@@ -88,6 +161,40 @@ pub trait InterchainTokenServiceInterface: AxelarExecutableInterface {
         gas_token: Token,
     ) -> Result<BytesN<32>, ContractError>;
 
+    /// Registers a canonical token as an interchain token.
+    ///
+    /// # Arguments
+    /// - `token_address` - The address of the canonical token.
+    ///
+    /// # Returns
+    /// - `Ok(BytesN<32>)`: Returns the token ID.
+    ///
+    /// # Errors
+    /// - `ContractError::TokenAlreadyRegistered`: If the token ID is already registered.
+    fn register_canonical_token(
```

Comment:
Move before the deploy_remote_canonical method since it's a more natural ordering